### PR TITLE
AT-1644: Add Google Cloud Storage for Pygyver

### PR DESCRIPTION
**Pull request by EvgeniyaRakina**
- CloudStorageExecutor class with several methods added

### CODE REVIEW

File: pygyver/etl/storage.py [hunk 1]
```diff
@@ -181,3 +181,115 @@ def read_json_files(self, prefix: str):
...
+        gcs_path: path to file in GCS
+        gcs_bucket: the GCS bucket, if None uses the GCS_BUCKET env
+        """
+        check_gcs_bucket(self, gcs_bucket=gcs_bucket)
+        self.bucket.blob(gcs_path).download_to_filename('temp_gcs_file.csv')
+        df = pd.read_csv('temp_gcs_file.csv')
+        os.remove("temp_gcs_file.csv")
+        return df
+
+    def gcs_list_files(self,
```
Comment (KarimZaoui) [inline pygyver/etl/storage.py:272]: `list_files`

File: pygyver/etl/storage.py [hunk 2]
```diff
@@ -181,3 +181,115 @@ def read_json_files(self, prefix: str):
...
+        """
+        bq_token_file_valid()
+        self.credentials = service_account.Credentials.from_service_account_file(
+            bq_token_file_path()
+        )
+        self.client = storage.Client(
+            credentials=self.credentials,
+            project=self.project_id
+        )
+        self.bucket = self.client.get_bucket(self.bucket_name)
```
Comment (KarimZaoui) [inline pygyver/etl/storage.py:213]: Remove bucket from auth

File: pygyver/etl/storage.py [hunk 3]
```diff
@@ -181,3 +181,115 @@ def read_json_files(self, prefix: str):
...
+
```
Comment (KarimZaoui) [inline pygyver/etl/storage.py:214]: New method:

def set_bucket(gcs_bucket=defaut_bucket()):
      self.bucket = self.client.get_bucket(gcs_bucket)

File: pygyver/etl/storage.py [hunk 4]
```diff
@@ -181,3 +181,115 @@ def read_json_files(self, prefix: str):
...
+                  gcs_bucket=None,
+                  index=False):
+        """
+        Writes pandas Dataframe to csv on GCS
+        gcs_path: path to file in GCS
+        df: pandas Dataframe to be uploaded
+        gcs_bucket: the GCS bucket, if None uses the GCS_BUCKET env
+        index: keeps the Dataframe index in file
+        """
+        check_gcs_bucket(self, gcs_bucket=gcs_bucket)
```
Comment (KarimZaoui) [inline pygyver/etl/storage.py:227]: self.set_bucket(gcs_bucket=gcs_bucket) instead of check_gcs_bucket

File: pygyver/etl/lib.py [hunk 5]
```diff
@@ -149,3 +149,24 @@ def s3_default_root():
...
+    return os.environ.get('GCS_BUCKET', '')
+
+def check_gcs_bucket(cse, gcs_bucket):
+    """
+    Checks if GCS bucket provided. If not, set GCS_BUCKET env as GCS bucket
+    """
+    if gcs_bucket is None:
+        cse.bucket = cse.client.get_bucket(self.bucket_name)
+    else:
+        cse.bucket = cse.client.get_bucket(gcs_bucket)
```
Comment (KarimZaoui) [inline pygyver/etl/lib.py:172]: Not needed anymore

File: pygyver/etl/lib.py [hunk 6]
```diff
@@ -149,3 +149,24 @@ def s3_default_root():
 
 def extract_args(batch_content, to_extract: str):
     return [x.get(to_extract, '') for x in batch_content if x.get(to_extract, '') != '']
+
+def gcs_default_project():
+    """
+    Returns GOOGLE_CLOUD_PROJECT if env is set
+    """
+    return os.environ.get('GOOGLE_CLOUD_PROJECT', '')
```
Comment (KarimZaoui) [inline pygyver/etl/lib.py:157]: Maybe STORAGE_PROJECT?

File: setup.py [hunk 7]
```diff
@@ -35,6 +35,7 @@ def read(*filenames, **kwargs):
         'gspread==3.1.0',
         'google-api-python-client==1.7.11',
         'oauth2client==4.1.3',
+        'google-cloud-storage==1.28.0'
```
Comment (bluefloyd00) [inline setup.py:38]: missing comma

**Pull request merged**